src/pfs/server/server_test.go

```diff
@@ -201,6 +209,393 @@ func TestDisallowReadsDuringCommit(t *testing.T) {
+	require.Equal(t, len(fileContent2), int(fileInfo.SizeBytes))
+
+	fileInfo, err = pfsutil.InspectFile(pfsClient, repo, commit2.ID, "foo", "", nil)
+	require.NoError(t, err)
+	require.Equal(t, commit2, fileInfo.CommitModified)
+	require.Equal(t, pfs.FileType_FILE_TYPE_REGULAR, fileInfo.FileType)
+	require.Equal(t, len(fileContent1) + len(fileContent2), int(fileInfo.SizeBytes))
+}
+
+func TestListFile(t *testing.T) {
+	t.Parallel()
+	pfsClient, _ := getClientAndServer(t)
+
+	repo := "test"
+	require.NoError(t, pfsutil.CreateRepo(pfsClient, repo))
+
+	commit, err := pfsutil.StartCommit(pfsClient, repo, "", "")
+	require.NoError(t, err)
+
+	fileContent1 := "foo\n"
+	_, err = pfsutil.PutFile(pfsClient, repo, commit.ID, "dir/foo", 0, strings.NewReader(fileContent1))
+	require.NoError(t, err)
+
+	fileContent2 := "bar\n"
+	_, err = pfsutil.PutFile(pfsClient, repo, commit.ID, "dir/bar", 0, strings.NewReader(fileContent2))
+	require.NoError(t, err)
+
+	require.NoError(t, pfsutil.FinishCommit(pfsClient, repo, commit.ID))
+
+	fileInfos, err := pfsutil.ListFile(pfsClient, repo, commit.ID, "dir", "", nil)
+	require.NoError(t, err)
+	require.Equal(t, 2, len(fileInfos))
+	require.True(t, fileInfos[0].File.Path == "dir/foo" && fileInfos[1].File.Path == "dir/bar" || fileInfos[0].File.Path == "dir/bar" && fileInfos[1].File.Path == "dir/foo")
+}
+
+func TestDeleteFile(t *testing.T) {
+	t.Parallel()
+	pfsClient, _ := getClientAndServer(t)
+
+	repo := "test"
+	require.NoError(t, pfsutil.CreateRepo(pfsClient, repo))
+
+	commit, err := pfsutil.StartCommit(pfsClient, repo, "", "")
+	require.NoError(t, err)
+
+	fileContent1 := "foo\n"
+	_, err = pfsutil.PutFile(pfsClient, repo, commit.ID, "foo", 0, strings.NewReader(fileContent1))
+	require.NoError(t, err)
+
+	fileContent2 := "bar\n"
+	_, err = pfsutil.PutFile(pfsClient, repo, commit.ID, "bar", 0, strings.NewReader(fileContent2))
+	require.NoError(t, err)
+
+	require.NoError(t, pfsutil.FinishCommit(pfsClient, repo, commit.ID))
+
+	// Because it's NO-OP right now
+	require.YesError(t, pfsutil.DeleteFile(pfsClient, repo, commit.ID, "foo"))
+
+	fileInfos, err := pfsutil.ListFile(pfsClient, repo, commit.ID, "", "", nil)
+	require.NoError(t, err)
+	require.Equal(t, 2, len(fileInfos))
+}
+
+func TestListCommit(t *testing.T) {
+	t.Parallel()
+	pfsClient, _ := getClientAndServer(t)
+
+	repo := "test"
+	require.NoError(t, pfsutil.CreateRepo(pfsClient, repo))
+
+	commit, err := pfsutil.StartCommit(pfsClient, repo, "", "")
+	require.NoError(t, err)
+
+	fileContent1 := "foo\n"
+	_, err = pfsutil.PutFile(pfsClient, repo, commit.ID, "foo", 0, strings.NewReader(fileContent1))
+	require.NoError(t, err)
+
+	require.NoError(t, pfsutil.FinishCommit(pfsClient, repo, commit.ID))
+
+	commitInfos, err := pfsutil.ListCommit(pfsClient, []string{repo})
+	require.NoError(t, err)
+	require.Equal(t, 1, len(commitInfos))
+
+	// test the block behaviour
+	ch := make(chan interface{})
```

Comment:
Throughout our code in other places I normally like to use `chan bool` for purpose like this since it's a little less typing and then call `close(ch)` rather than sending a value.
The first part is fairly superfluous but I think a consistent convention is good.
The second part of using close rather than sending is because it's really easy to accidentally leak a goroutine if never recv from the channel. Or worse get yourself in a dead lock.

You're not actually hitting either of those problems here but I think we should all just get in the habit of using `chan bool` and `close` for stuff like this.